engine/access/rpc/rate_limit_interceptor.go

```diff
@@ -0,0 +1,110 @@
+package rpc
+
+import (
+	"context"
+	"path/filepath"
+
+	"github.com/rs/zerolog"
+	"golang.org/x/time/rate"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// TODO: read this map from an input file
```

Comment:
I believe we can use flags here which might be better so all config is in the same place

Eg:
```
--rate-limit-method "Ping=300"
```
https://pkg.go.dev/github.com/spf13/pflag#StringToInt